Provide more informative kernel names in Inductor

Chillee opened the pull request:
Stack from [ghstack](https://github.com/ezyang/ghstack) (oldest at bottom):
* #96170
* __->__ #95940

Before: `triton_fused_add_83_add_84_relu_13_squeeze_46_var_mean_15_14`
After: `triton_fused__native_batch_norm_legit_functional_convolution_relu_14`

cc @soumith @voznesenskym @yanboliang @penguinwu @anijain2305 @EikanWang @jgong5 @Guobing-Chen @XiaobingSuper @zhuhaozhe @blzheng @Xia-Weiwen @wenzhe-nrv @jiayisunx @peterbell10 @desertfire

Chillee committed:
Provide more informative kernel names in Inductor

[ghstack-poisoned]

Chillee committed:
Update on "Provide more informative kernel names in Inductor"




cc soumith voznesenskym yanboliang penguinwu anijain2305 EikanWang jgong5 Guobing-Chen XiaobingSuper zhuhaozhe blzheng Xia-Weiwen wenzhe-nrv jiayisunx peterbell10 desertfire

[ghstack-poisoned]

SherlockNoMad commented on torch/fx/proxy.py:
```diff
@@ -164,19 +164,15 @@ def create_proxy(self, kind: str, target: Target, args: Tuple[Any, ...], kwargs:
 
-            # Explicitly set the stack_trace, nn_module_stack and source_fn on the node.meta
-            # If other meta fields are needed, they can be added here
             stack_trace = current_meta.get("stack_trace")
```
nit, stack_trace can also be included in the copy_meta_fields

Chillee replied:
Stack trace is stored as an attribute directly on node, while the other metadata are stored in the meta field.

SherlockNoMad commented on torch/fx/experimental/proxy_tensor.py:
```diff
@@ -471,6 +472,23 @@ def wrapped(*proxies):
+    global ORIGINAL_ATEN
+    if ORIGINAL_ATEN is None:
+        ORIGINAL_ATEN = func
+        fx_traceback.current_meta['original_aten'] = func
```
condition on 
fx_traceback.has_preserved_node_meta() is True

SherlockNoMad commented:
Need some test. see
test_transformer_preserves_nn_module_stack_for_get_attr()  in  test_fx.py 
test_export_with_stack_trace() in test_export.py

SherlockNoMad commented on torch/_dynamo/output_graph.py:
```diff
@@ -810,6 +810,9 @@ def create_proxy(
             rv.node.meta["source_fn"] = target
+        elif kind == "call_module":
+            # For modules we store the class
+            rv.node.meta["source_fn"] = rv.node.meta["nn_module_stack"][target][1]
```
cc @jerryzh168

SherlockNoMad reviewed (approved):
per comment

Chillee committed:
Update on "Provide more informative kernel names in Inductor"


Before: `triton_fused_add_83_add_84_relu_13_squeeze_46_var_mean_15_14`
After: `triton_fused__native_batch_norm_legit_functional_convolution_relu_14`

cc soumith voznesenskym yanboliang penguinwu anijain2305 EikanWang jgong5 Guobing-Chen XiaobingSuper zhuhaozhe blzheng Xia-Weiwen wenzhe-nrv jiayisunx peterbell10 desertfire

[ghstack-poisoned]

Chillee committed:
Update on "Provide more informative kernel names in Inductor"


Before: `triton_fused_add_83_add_84_relu_13_squeeze_46_var_mean_15_14`
After: `triton_fused__native_batch_norm_legit_functional_convolution_relu_14`

cc soumith voznesenskym yanboliang penguinwu anijain2305 EikanWang jgong5 Guobing-Chen XiaobingSuper zhuhaozhe blzheng Xia-Weiwen wenzhe-nrv jiayisunx peterbell10 desertfire

[ghstack-poisoned]

Chillee committed:
Update on "Provide more informative kernel names in Inductor"


Before: `triton_fused_add_83_add_84_relu_13_squeeze_46_var_mean_15_14`
After: `triton_fused__native_batch_norm_legit_functional_convolution_relu_14`

cc soumith voznesenskym yanboliang penguinwu anijain2305 EikanWang jgong5 Guobing-Chen XiaobingSuper zhuhaozhe blzheng Xia-Weiwen wenzhe-nrv jiayisunx peterbell10 desertfire

[ghstack-poisoned]

Chillee committed:
Update on "Provide more informative kernel names in Inductor"


Before: `triton_fused_add_83_add_84_relu_13_squeeze_46_var_mean_15_14`
After: `triton_fused__native_batch_norm_legit_functional_convolution_relu_14`

cc soumith voznesenskym yanboliang penguinwu anijain2305 EikanWang jgong5 Guobing-Chen XiaobingSuper zhuhaozhe blzheng Xia-Weiwen wenzhe-nrv jiayisunx peterbell10 desertfire

[ghstack-poisoned]

Chillee committed:
Update on "Provide more informative kernel names in Inductor"


Before: `triton_fused_add_83_add_84_relu_13_squeeze_46_var_mean_15_14`
After: `triton_fused__native_batch_norm_legit_functional_convolution_relu_14`

cc soumith voznesenskym yanboliang penguinwu anijain2305 EikanWang jgong5 Guobing-Chen XiaobingSuper zhuhaozhe blzheng Xia-Weiwen wenzhe-nrv jiayisunx peterbell10 desertfire

[ghstack-poisoned]

Chillee commented:
@pytorchbot merge

Chillee committed:
Update on "Provide more informative kernel names in Inductor"


Before: `triton_fused_add_83_add_84_relu_13_squeeze_46_var_mean_15_14`
After: `triton_fused__native_batch_norm_legit_functional_convolution_relu_14`

cc soumith voznesenskym yanboliang penguinwu anijain2305 EikanWang jgong5 Guobing-Chen XiaobingSuper zhuhaozhe blzheng Xia-Weiwen wenzhe-nrv jiayisunx peterbell10 desertfire

[ghstack-poisoned]

Chillee commented:
@pytorchbot merge

Chillee commented:
@pytorchbot merge -f "seems unlikely that accuracy issues would be caused by this PR"
